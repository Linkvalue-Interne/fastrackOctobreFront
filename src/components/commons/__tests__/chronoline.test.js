import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import * as emotion from 'emotion';
import { createSerializer } from 'jest-emotion'
import ChronoLine, { Ellipse, Line } from '../chronoline';


expect.addSnapshotSerializer(createSerializer(emotion));
configure({ adapter: new Adapter() });

describe('ChronoLine', () => {
  it('should renders Line with correct styles', () => {
    const mock = {
      colors: {
        primary: '#111',
      },
    };
    const tree = renderer.create(<Line theme={mock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders Ellipse with correct styles', () => {
    const mock = {
      colors: {
        primary: '#111',
      },
    };
    const tree = renderer.create(<Ellipse theme={mock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders ChronoLine with correct styles', () => {

    const theme2 = {
      colors: {
        primary: '#111',
      },
    };

    // const shallowWithTheme = (tree, theme) => {
    //   const context = shallow(<ThemeProvider theme={theme} />)
    //     .instance()
    //     .getChildContext();
    //   return shallow(tree, { context });
    // };

    // const tree = shallowWhithTheme(<ChronoLine />, theme2).toJSON();
    const wrapper = shallow(<ChronoLine theme={theme2} />)
    expect(wrapper).toMatchSnapshot();
  });
});
