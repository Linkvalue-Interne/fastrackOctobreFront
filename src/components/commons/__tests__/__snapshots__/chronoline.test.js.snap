// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChronoLine should renders ChronoLine with correct styles 1`] = `ShallowWrapper {}`;

exports[`ChronoLine should renders Ellipse with correct styles 1`] = `
.emotion-0 {
  position: absolute;
  width: 1rem;
  height: 1rem;
  border-radius: 25px;
  top: 50%;
  -ms-transform: translateY(-50%);
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<div
  className="emotion-0"
/>
`;

exports[`ChronoLine should renders Line with correct styles 1`] = `
.emotion-0 {
  margin: 2rem 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 5px;
  background: #111;
}

<div
  className="emotion-0"
/>
`;
